i6_models/parts/frontend/vgg.py

```diff
@@ -0,0 +1,384 @@
+from __future__ import annotations
+
+__all__ = [
+    "VGG4LayerActFrontendV1",
+    "VGG4LayerActFrontendV1Config",
+    "VGG4LayerPoolFrontendV1",
+    "VGG4LayerPoolFrontendV1Config",
+]
+
+from dataclasses import dataclass
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+
+IntTupleIntType = Union[int, Tuple[int, int]]
+
+
+@dataclass
+class VGG4LayerActFrontendV1Config(ModelConfiguration):
+    """
+    Attributes:
+        conv1_channels: number of channels for first conv layer
+        conv2_channels: number of channels for second conv layer
+        conv3_channels: number of channels for third conv layer
+        conv4_channels: number of channels for fourth conv layer
+        conv_kernel_size: kernel size of conv layers
+        conv_padding: padding for the convolution
+        pool1_kernel_size: kernel size of first pooling layer
+        pool1_stride: stride of first pooling layer
+        pool1_padding: padding for first pooling layer
+        pool2_kernel_size: kernel size of second pooling layer
+        pool2_stride: stride of second pooling layer
+        pool2_padding: padding for second pooling layer
+        activation: activation function at the end
+        linear_input_dim: input size of the final linear layer
+        linear_output_dim: output size of the final linear layer
+    """
+
+    conv1_channels: int
+    conv2_channels: int
+    conv3_channels: int
+    conv4_channels: int
+    conv_kernel_size: IntTupleIntType
+    conv_padding: Optional[IntTupleIntType]
+    pool1_kernel_size: IntTupleIntType
+    pool1_stride: Optional[IntTupleIntType]
+    pool1_padding: Optional[IntTupleIntType]
+    pool2_kernel_size: IntTupleIntType
+    pool2_stride: Optional[IntTupleIntType]
+    pool2_padding: Optional[IntTupleIntType]
+    activation: Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]
+    linear_input_dim: Optional[int]
```

Comment:
Does it really make sense to have this configurable? There is only one single valid value for it, which is via the flattening of the layer before.

Reply:
this is supposed to be an implicit user check.. this is what the user expects as output size from the front end. I prefer to leave it in.